utils/python/update_acme_tests.py

```diff
@@ -114,9 +114,18 @@ def get_test_suite(suite, machine=None, compiler=None):
     Return a list of FULL test names for a suite.
     """
     expect(suite in _TEST_SUITES, "Unknown test suite: '%s'" % suite)
-
-    machine = cime_util.probe_machine_name() if machine is None else machine
-    compiler = cime_util.get_machine_info("COMPILER", machine=machine)[0] if compiler is None else compiler
+    machobj = Machines()
+    if(machine is None):
+        machobj.probe_machine_name()
+    else:
+        machobj.set_machine(machine)
+    if(compiler is None):
+        compiler = machobj.get_default_compiler()
+    else:
+        if(machobj.is_valid_compiler(compiler)):
```

Comment:
This could be slightly simpler as:
expect(machobj.is_valid_compiler(compiler) ,"Compiler %s not supported for machine %s" %(compiler,machine))
machobj.set_value('COMPILER',compiler)